samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -74,7 +74,7 @@ public GroupByContainerCount(int containerCount) {
     // Convert to a Set of ContainerModel
     Set<ContainerModel> containerModels = new HashSet<>();
     for (int i = 0; i < containerCount; i++) {
-      containerModels.add(new ContainerModel(i, taskGroups[i]));
+      containerModels.add(new ContainerModel(String.valueOf(i), taskGroups[i]));
```

Comment:
@jmakes  This grouper essentially relies on job.container.count. It "generates" containerId according to the count and creates ContainerModels. There is no question of not being able to parse UUIDs. So, I am slightly lost of where the "external" processor ids are even passed in and where to add the validation. Let's discuss this in person and sort it out.